addons/udes_stock/models/stock_quant_package.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from collections import defaultdict
+
+
+class StockQuantPackage(models.Model):
+    _name = 'stock.quant.package'
+    _inherit = ['stock.quant.package', 'abstract.stock.model']
+
+    # Enable create packages
+    ASM_CREATE = True
+
+    def _get_product_quantities_by_key(self, get_key=lambda q: q.product_id):
+        """ This function computes the product quantities for the given package grouped by a key
+            :args:
+                get_key: a callable which takes a quant and returns the key
+        """
+        return self._get_contained_quants().group_quantity_by_product(get_key=get_key)
+
+    def mls_can_fulfil(self, mls):
+        """ Returns mls which the package can fulfil. If the product_qty of the
+            mls is larger than in the package (i.e. in self) the mls will be split.
+            :args:
+                mls: move lines to check against
+            :returns: the move lines the package can fulfil, and those split out that it can't
+        """
+        MoveLines = self.env['stock.move.line']
+        pack_quantities = self._get_all_products_quantities()
+        can_fulfil_mls = MoveLines.browse()
+        excess_mls = MoveLines.browse()
+        for prod, mls_grp in mls.groupby('product_id'):
+            pack_qty = pack_quantities.get(prod, 0)
+            if pack_qty == 0:
+                # just skip over
+                continue
+            fulfil_mls, excess_ml, _ = mls_grp.move_lines_for_qty(pack_qty)
+            can_fulfil_mls |= fulfil_mls
+            if excess_ml:
+                excess_mls |= excess_ml
+        return can_fulfil_mls, excess_mls
+
+    def has_same_content(self, other, get_key=lambda x: x.items()):
+        """ Compare the content of current package with the content of another package
+            :args:
+                - other: Other package to compare with
+            :kwargs:
+                - get_key: what to compare, dictionary method
+            :returns:
+                - boolean flag if two packages are the same or not (based on comparison method)
+        """
+        self.ensure_one()
+        return frozenset(get_key(self._get_all_products_quantities())) == frozenset(
+            get_key(other._get_all_products_quantities())
+        )
```

Comment:
I think the initial idea was to have more flexibility and be able to check if they have only the same products without checking their quantities, etc
Since that can be done with a simple `self.product_id == other.product_id`, I am changing get_key to be used for what you have suggested.

Reply:
```get_compare_values``` or something like that?